src/history/hs_rec.c

```diff
@@ -113,6 +125,8 @@ __hs_insert_record(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_BTREE *btree,
     cursor->set_key(cursor, 4, btree->id, key, tw->start_ts, UINT64_MAX);
     WT_ERR_NOTFOUND_OK(__wt_curhs_search_near_before(session, cursor), true);
 
+    F_CLR(cursor, WT_CURSTD_HS_READ_ALL);
+
```

Comment:
I think we should restore the previous flag setting instead of always clearing the flag.

Reply:
Fixed.